docs/src/userguide/extension_types.rst

```diff
@@ -1051,5 +1051,27 @@ generated containing declarations for its object struct and type object. By
 including the ``.h`` file in external C code that you write, that code can
 access the attributes of the extension type.
 
+Dataclass extension types
+=========================
```

Comment:
Currently `@cython.dataclasses.dataclass` requires it to be a `cdef class` but doesn't change it to one (that applies in both regular mode and pure-Python mode).

I'm not convinced it should be assumed to be an extension type. I see why it might be nice, but for now I think it might be better left explicit.

Reply:
The reasoning being that if someone wanted a normal dataclass even when cythonized, they would use `@dataclasses.dataclass`. I can agree on the explicitness though this is good.

Could you make a note of the answers to my questions in the documentation?